src/server/rdb_save.cc

```diff
@@ -433,6 +433,9 @@ error_code RdbSerializer::SaveListObject(const PrimeValue& pv) {
         RETURN_ON_ERR(SaveLzfBlob(Bytes{reinterpret_cast<uint8_t*>(data), compress_len}, node->sz));
       } else {
         RETURN_ON_ERR(SaveString(node->entry, node->sz));
+        if (!node->next) {
+          is_last_entry_ = true;
+        }
         FlushIfNeeded();
```

Comment:
How about we pass some bit to `FlushIfNeeded()`, indicating whether this is mid value or not?

Reply:
which would need to propagate down the call chain until `PrepareFlush` :( that's the reason I made it a member/flag

Reply:
it's less error prone to pass a parameter and be stateless..
is it so much passing around?
you could also accept a default value if that helps

Reply:
The default won't help cause the methods are virtual and default parameters on virtual functions work with the static type of the object which is error prone. However because I did had a bug around the state I will apply the comment even if it means passing the bool around. At least it gurantees correctness